core/providers/java/java.go

```diff
@@ -0,0 +1,87 @@
+package java
+
+import (
+	"fmt"
+
+	"github.com/railwayapp/railpack/core/generate"
+	"github.com/railwayapp/railpack/core/plan"
+)
+
+type JavaProvider struct{}
+
+func (p *JavaProvider) Name() string {
+	return "java"
+}
+
+func (p *JavaProvider) Detect(ctx *generate.GenerateContext) (bool, error) {
+	return ctx.App.HasMatch("pom.{xml,atom,clj,groovy,rb,scala,yaml,yml}") || ctx.App.HasMatch("gradlew"), nil
+}
+
+func (p *JavaProvider) Initialize(ctx *generate.GenerateContext) error {
+	return nil
+}
+
+func (p *JavaProvider) StartCommandHelp() string {
+	return ""
+}
+
+func (p *JavaProvider) Plan(ctx *generate.GenerateContext) error {
+
+	build := ctx.NewCommandStep("build")
+	build.AddCommand(plan.NewCopyCommand("."))
+	build.Inputs = []plan.Input{plan.NewStepInput(ctx.GetMiseStepBuilder().Name())}
+
+	if p.usesGradle(ctx) {
+		p.setGradleVersion(ctx)
+		p.setJDKVersion(ctx, nil)
+
+		if ctx.App.HasMatch("gradlew") && !ctx.App.IsFileExecutable("gradlew") {
+			build.AddCommand(plan.NewExecCommand("chmod +x gradlew"))
+		}
+
+		build.AddCommand(plan.NewExecCommand("./gradlew clean build -x check -x test"))
+		build.AddCache(p.gradleCache(ctx))
+	} else {
+		ctx.GetMiseStepBuilder().Default("maven", "latest")
+		p.setJDKVersion(ctx, nil)
+
+		if ctx.App.HasMatch("mvnw") && !ctx.App.IsFileExecutable("mvnw") {
+			build.AddCommand(plan.NewExecCommand("chmod +x mvnw"))
+		}
+
+		build.AddCommand(plan.NewExecCommand(fmt.Sprintf("%s -DoutputFile=target/mvn-dependency-list.log -B -DskipTests clean dependency:list install", p.getMavenExe(ctx))))
+	}
+
+	runtimeDeps := ctx.NewMiseStepBuilder("mise:install-runtime")
+	runtimeDeps.Inputs = []plan.Input{ctx.DefaultRuntimeInput()}
```

Comment:
I don't think this does anything. The inputs for the mise step should not really be changed since it needs the a specific input to get the mise binary